product_set/models/product_set.py

```diff
@@ -52,3 +54,15 @@ def _compute_display_name(self):
             if rec.partner_id and rec.partner_id.name:
                 parts.append(f"@ {rec.partner_id.name}")
             rec.display_name = " ".join(map(str, parts))
+
+    @api.model
+    def _search_display_name(self, operator, value):
+        domain = [
+            "|",
+            "|",
+            ("name", operator, value),
+            ("ref", operator, value),
+            ("partner_id.name", operator, value),
+        ]
+
+        return domain
```

Comment:
You can compact the code:

```suggestion
        return [
            "|",
            "|",
            ("name", operator, value),
            ("ref", operator, value),
            ("partner_id.name", operator, value),
        ]
```